template/src/components/RTMConfigure.tsx

```diff
@@ -107,57 +130,349 @@ const RtmConfigure = (props: any) => {
+        }
       };
-
-      if (!isWebInternal()) return;
-      window.addEventListener(
-        'beforeunload',
-        isWeb() && !isSDK() ? handBrowserClose : () => {},
-      );
-
+      // Set up window listeners
+      if (isWeb() && !isSDK()) {
+        window.addEventListener('beforeunload', handBrowserClose);
+      }
       window.addEventListener('pagehide', logoutRtm);
-      // cleanup this component
+
+      // Remove listeners on unmount
       return () => {
-        window.removeEventListener(
-          'beforeunload',
-          isWeb() && !isSDK() ? handBrowserClose : () => {},
-        );
         window.removeEventListener('pagehide', logoutRtm);
+        if (isWeb() && !isSDK()) {
+          window.removeEventListener(
+            'beforeunload',
+            isWeb() && !isSDK() ? handBrowserClose : () => {},
+          );
+        }
       };
     }
   }, []);
 
+  const init = async (rtcUid: UidType) => {
+    //on sdk due to multiple re-render we are getting rtm error code 8
+    //you are joining the same channel too frequently, exceeding the allowed rate of joining the same channel multiple times within a short period
+    //so checking rtm connection state before proceed
+
+    // Check if already connected (equivalent to v1.5x connectionState === 'CONNECTED')
+    if (rtmConnectionState === 2 && RTMEngine.getInstance().isEngineReady) {
+      logger.log(
+        LogSource.AgoraSDK,
+        'Log',
+        'RTM already connected, skipping initialization',
+      );
+      return;
+    }
+
+    logger.log(LogSource.AgoraSDK, 'Log', 'RTM creating engine...');
+
+    try {
+      if (!RTMEngine.getInstance().isEngineReady) {
+        RTMEngine.getInstance().setLocalUID(rtcUid);
+        logger.log(LogSource.AgoraSDK, 'API', 'RTM local Uid set');
+      }
+      engine.current = RTMEngine.getInstance().engine;
+    } catch (error) {
+      logger.error(
+        LogSource.AgoraSDK,
+        'Log',
+        'RTM engine initialization failed:',
+        error,
+      );
+      throw error;
+    }
+    // engine.current = createAgoraRtmClient(
+    //   new RtmConfig({
+    //     userId: uid,
+    //     appId: $config.APP_ID,
+    //     useStringUserId: true,
+    //   }),
+    // );
+    logger.log(LogSource.AgoraSDK, 'Log', 'RTM engine creation done');
+
+    engine.current.addEventListener(
+      'linkState',
+      async (data: LinkStateEvent) => {
+        // Update connection state for duplicate initialization prevention
+        setRtmConnectionState(data.currentState);
+
+        // Log connection state changes (equivalent to v1.5x connectionStateChanged)
+        logger.log(
+          LogSource.AgoraSDK,
+          'Event',
+          `RTM linkState changed: ${data.previousState} -> ${data.currentState}`,
+          data,
+        );
+
+        // Handle connection errors (equivalent to v1.5x error events)
+        if (data.currentState === 5) {
```

Comment:
Please use the enum instead of number comparison. so it will be easy to understand